SignalServiceKit/src/Storage/KeychainStorage.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import SAMKeychain
+
+public enum KeychainStorageError: Error {
+    case failure(description: String)
+}
+
+// MARK: -
+
+@objc public protocol KeychainStorage: class {
+
+    @objc func string(forKey key: String, service: String) throws -> String
+
+    @objc func set(string: String, forKey key: String, service: String) throws
+
+    @objc func data(forKey key: String, service: String) throws -> Data
+
+    @objc func set(data: Data, forKey key: String, service: String) throws
+
+    @objc func remove(key: String, service: String) throws
+}
+
+// MARK: -
+
+@objc
+public class SSKKeychainStorage: NSObject, KeychainStorage {
+
+    @objc public static let sharedInstance = SSKKeychainStorage()
+
+    // Force usage as a singleton
+    override private init() {
+        super.init()
+
+        SwiftSingletons.register(self)
+    }
+
+    @objc public func string(forKey key: String, service: String) throws -> String {
+        var error: NSError?
+        let result = SAMKeychain.password(forService: service, account: key, error: &error)
```

Comment:
huh - I'm surprised this didn't get interpreted by Swift as a throwable method, but I'm glad our wrapper throws.